forge-core/src/main/java/forge/item/PaperCardPredicates.java

```diff
@@ -76,24 +76,20 @@ private PredicatePrintedWithRarity(final CardRarity rarity) {
     }
 
     private static final class PredicateColor implements Predicate<PaperCard> {
-        private final byte operand;
+        private final MagicColor.Color operand;
 
-        private PredicateColor(final byte color) {
+        private PredicateColor(final MagicColor.Color color) {
             this.operand = color;
         }
 
         @Override
         public boolean test(final PaperCard card) {
-            for (final byte color : card.getRules().getColor()) {
-                if (color == operand) {
-                    return true;
-                }
+            if (card.getRules().getColor().toEnumSet().contains(operand)) {
+                return true;
             }
             if (card.getRules().getType().hasType(CardType.CoreType.Land)) {
-                for (final byte color : card.getRules().getColorIdentity()) {
-                    if (color == operand) {
-                        return true;
-                    }
+                if (card.getRules().getColorIdentity().toEnumSet().contains(operand)) {
```

Comment:
what's wrong with iterating?
creating new Set from List objects each time seems like a pretty bad refactor performance wise when filtering larger PaperCard pools 👎

Reply:
i wanted to remove Iterable<byte> in favor for iterable<Enum> in later future